src/exercicio-2.ts

```diff
@@ -0,0 +1,84 @@
+  {"id" : 4, "name": "Nicolau Copérnico", "bio": "Nicolau Copérnico foi um astrônomo e matemático polonês que desenvolveu a teoria heliocêntrica do Sistema Solar."}];
+
+
+/**
+ * Encontra um objeto Scientist na lista através do id e retorna a bio do mesmo.
+ * @param list Array de Scientists declarado acima.
+ * @param id Id usado para acessar cada objeto Scientist no Array list.
+ * @returns Uma string, que será a bio do objeto Scientist encontrado, caso contrário "Nenhum cientista encontrado".
+ */
+const getScientistBio = (list: Scientist[], id: number): string => {
+  let scientist: Scientist | undefined = list.find(s => s.id === id);
+  return scientist ? scientist.bio : "Nenhum cientista encontrado";
+}
+
+console.log(getScientistBio(lista, 3))
+
+/**
+ * Encontra um objeto Scientist na lista através do id e apaga o mesmo da lista.
+ * @param list Array de Scientists declarado acima.
+ * @param id Id usado para acessar cada objeto Scientist no Array list.
+ * @returns Uma string, que será o atributo nome do objeto Scientist encontrado, caso contrário, "Nenhum cientista encontrado".
+ */
+const getScientistName = (list: Scientist[], id: number): string => {
+  let scientist: Scientist | undefined = list.find(s => s.id === id);
+  return scientist ? scientist.name : "Nenhum cientista encontrado";
+}
+
+console.log(getScientistName(lista, 3))
+
+/**
+ * Apaga um objeto Scientist da lista através do id.
+ * @param list Array de Scientists declarado acima.
+ * @param id Id usado para acessar cada objeto Scientist no Array list.
+ * @returns A lista 'list', atualizada, já sem o objeto Scientist que foi removido ou, caso o
+ * id não seja encontrado na lista, será retornada a lista original.
+ */
+const deleteScientist = (list: Scientist[], id: number): Scientist[] => {
+  let scientist: Scientist | undefined = list.find(s => s.id === id);
+  scientist ? list.splice(list.indexOf(scientist), 1) : list
+  return list
+}
+
+console.log(deleteScientist(lista, 10))
+
+/**
+ * Atualiza o nome ou a bio de um objeto Scientist da lista através do id.
+ * @param list Array de Scientists declarado acima.
+ * @param id Id usado para acessar cada objeto Scientist no Array list.
+ * @param name String que será inserida como valor do atributo nome do objeto Scientist selecionado.
+ * @param bio String que será inserida como valor do atributo bio do objeto Scientist selecionado.
+ * @returns O objeto Scientist atualizado, caso tenha sido alterado nome e/ou bio. Caso contrário retorna
+ * uma string "Nenhum cientista encontrado".
+ */
+const updateScientist = (list: Scientist[], id: number, name: string, bio: string): Scientist | string => {
+  let scientistToUpdate: Scientist | undefined = list.find(s => s.id === id);
```

Comment:
Quando falamos de abordagem funcional, estamos falando em dizer passo a passo, por exemplo, na abordagem imperativa, não usariamos a função find, e sim utilizariamos alguma forma de for ( for in, for )

Também é preciso nomear de forma mais clara as variaveis, para ficar mais claro do que se trata.

Reply:
Fiz estas alterações. Espero que estejam a contento.